### PR TITLE
Optimize pdf

### PR DESCRIPTION
**Pull request by willhu-jpg**

### CODE REVIEW

File: Estimate Precision Visualization.ipynb [hunk 1]
```diff
@@ -190,7 +190,7 @@
    "source": [
     "def p_d(tx, dx, n, P_rx, P_tx, D_std):\n",
     "    modeledDistance = RSSItoDistance(P_rx, P_tx, n)\n",
-    "    return ss.norm.pdf(np.linalg.norm(dx - tx), loc=modeledDistance, scale=D_std)\n",
+    "    return math.exp(-(((np.linalg.norm(dx-tx)-modeledDistance)/D_std)**2)/2) / (math.sqrt(2*math.pi) * D_std)\n",
```
Comment (ntlhui) [inline Estimate Precision Visualization.ipynb]: Is there a reason that we are using math.exp, math.sqrt, and math.pi instead of their numpy equivalents?

I would say that we should add a Markdown cell above to write out the math a bit more explicitly, since it is a bit hard to follow as a one-liner.  You may want to try breaking apart the one-liner for readability, but only if it doesn't impact your runtime (it might not).

File: Estimate Precision Visualization.ipynb [hunk 2]
```diff
@@ -8,12 +8,12 @@
...
     "import numpy as np\n",
     "import scipy.optimize as so\n",
-    "import scipy.stats as ss\n",
     "import utm\n",
     "import datetime as dt\n",
     "import time\n",
     "import matplotlib.pyplot as plt\n",
-    "from matplotlib.patches import Circle\n"
+    "from matplotlib.patches import Circle\n",
+    "import math\n"
```
Comment (ntlhui) [inline Estimate Precision Visualization.ipynb:16]: See comment below re `numpy` vs `math`

**Review by ntlhui — CHANGES_REQUESTED** [review]
See changes regarding libraries and readability.

File: Estimate Precision Visualization.ipynb (hunk 1 above)
Comment (willhu-jpg) [reply Estimate Precision Visualization.ipynb]: I benchmarked math and the NumPy equivalents and found that math was faster for all operations and constants.
Did a bit of research online and found that math functions typically get replaced with an inline equivalent while the NumPy version remains as a function call, leading to the difference in runtime.

I also edited the Markdown cell right above to include the math. Let me know if I should change the format though.

Finally, I broke down the one-liner to calculate x separately.

File: Estimate Precision Visualization.ipynb [hunk 3]
```diff
@@ -190,7 +204,8 @@
    "source": [
     "def p_d(tx, dx, n, P_rx, P_tx, D_std):\n",
     "    modeledDistance = RSSItoDistance(P_rx, P_tx, n)\n",
-    "    return ss.norm.pdf(np.linalg.norm(dx - tx), loc=modeledDistance, scale=D_std)\n",
+    "    adjustedDistance = (np.linalg.norm(dx-tx)-modeledDistance)/D_std\n",
+    "    return math.exp((-(adjustedDistance**2)/2) / (math.sqrt(2*math.pi) * D_std)\n",
```
Comment (ntlhui) [inline Estimate Precision Visualization.ipynb]: Is there a significant performance difference between using `math` vs `np`?

File: Estimate Precision Visualization.ipynb [hunk 4]
```diff
@@ -179,7 +179,21 @@
...
+    "\n",
+    "$$f(x) = \\frac{\\exp(\\frac{-x^{2}}{2})}{D\\_std\\sqrt{2\\pi}}$$\n",
+    "\n",
+    "where x is described as\n",
+    "\n",
+    "$$x = \\frac{\\left|\\left|dx - tx\\right|\\right| - \\widehat{d}}{D\\_std}$$\n",
+    "\n",
+    "and $\\widehat{d}$ is equal to\n",
+    "\n",
+    "$$\\widehat{d} = 10 ^ {\\frac{P_{tx} - P_{{rx}_i}}{10n}}$$"
```
Comment (ntlhui) [inline Estimate Precision Visualization.ipynb]: Please check that your notation is consistent with the notation used earlier in this document.  It seems you have escaped backslashes.

**Review by ntlhui — CHANGES_REQUESTED** [review]
See comments

File: Estimate Precision Visualization.ipynb (hunk 1 above)
Comment (ntlhui) [reply Estimate Precision Visualization.ipynb]: ok

File: Estimate Precision Visualization.ipynb (hunk 3 above)
Comment (ntlhui) [reply Estimate Precision Visualization.ipynb]: See above

File: Estimate Precision Visualization.ipynb (hunk 2 above)
Comment (ntlhui) [reply Estimate Precision Visualization.ipynb:16]: Resolved (see above)

File: Estimate Precision Visualization.ipynb (hunk 4 above)
Comment (willhu-jpg) [reply Estimate Precision Visualization.ipynb]: fixed the notation. Let me know if there's anything else.

Comment (ntlhui) [reply Estimate Precision Visualization.ipynb]: Unknown what standard variation is.  Standard deviation of x is normally denoted as $\sigma_x$, variance is denoted as $\sigma^2_x$.  Note that we are using $l_{tx}$ to denote the transmitter location vector, $l_{rx}$ to denote the receiver location vector, and $\left|\vec{x}\right|$ to denote the l2 norm of a vector.

**Review by ntlhui — CHANGES_REQUESTED** [review]
See comment response

File: Estimate Precision Visualization.ipynb (hunk 4 above)
Comment (willhu-jpg) [reply Estimate Precision Visualization.ipynb]: Just updated the changes. Simplified the equation a bit, but it's still fairly dense.

Comment (ntlhui) [reply Estimate Precision Visualization.ipynb]: Notation corrected, using vector and \left\right pairing

**Review by ntlhui — APPROVED** [review]
